cudamapper/src/main.cu

```diff
@@ -154,102 +157,75 @@ int main(int argc, char* argv[])
     };
     std::future<void> overlap_result(std::async(std::launch::async, overlaps_writer_func));
 
+    auto overlapper = claragenomics::cudamapper::OverlapperTriggered();
+
     // Track overall time
     std::chrono::milliseconds index_time      = std::chrono::duration_values<std::chrono::milliseconds>::zero();
     std::chrono::milliseconds matcher_time    = std::chrono::duration_values<std::chrono::milliseconds>::zero();
     std::chrono::milliseconds overlapper_time = std::chrono::duration_values<std::chrono::milliseconds>::zero();
 
-    //Now carry out all the looped polling
-    //size_t query_start = 0;
-    //size_t query_end = query_start + index_size - 1;
-
-    for (size_t query_start = 0; query_start < queries; query_start += index_size)
+    for (size_t query_start_index = 0; query_start_index < queries; query_start_index += index_size)
     { // outer loop over query
-        size_t query_end = std::min(query_start + index_size, static_cast<size_t>(queries));
-        auto start_time  = std::chrono::high_resolution_clock::now();
-
-        //For every range of reads the process is to first generate all-vs-all overlaps
-        //for that chunk and then to generate its overlaps with subsequent chunks.
-        //For example, if a FASTA was chunked into 4 chunks: A,B,C,D the process would be as follows:
-        //
-        // Add overlaps for All-vs-all for chunk A
-        // Add overlaps for Chunk A vs Chunk B
-        // Add overlaps for Chunk A vs Chunk C
-        // Add overlaps for All-vs-all for chunk B
-        // Add overlaps for Chunk B vs Chunk C
-        // Add overlaps for All-vs-all for chunk C
-        std::pair<std::uint64_t, std::uint64_t> query_range{query_start, query_end};
-
-        auto overlapper = claragenomics::cudamapper::OverlapperTriggered();
-
-        size_t target_start = 0;
+        size_t query_one_past_the_end_index = std::min(query_start_index + index_size, static_cast<size_t>(queries));
```

Comment:
just call it `query_end_index` please. This is such a common pattern in C++, that the verbosity of _one_past_the_end should really not be necessary.

I've seen people use `_last` very often if they meant to include the element.

Reply:
This one is a really tricky to decide. I agree that to every C++ programmer it should be clear that container::end() is one_past_the_end, but `query_end_index` can mean both things, probably even leaning towards it being actual last valid index...

Reply:
done